app/src/main/java/com/example/pokedix/extensions/PokemonDetailsResponse.kt

```diff
@@ -0,0 +1,37 @@
+package com.example.pokedix.extensions
+
+import com.example.pokedix.models.*
+
+fun PokemonDetailsResponse.toPokemonDetails(
+    pokedexSelected: PokedexList,
+    pokemonSpecieResponse: PokemonSpecieResponse
+): PokemonDetails {
+    var type2 = PokemonType.NO
+    if (types.size > 1)
+        type2 = PokemonType.valueOf(types.last().type.name.toValidFormatTypesEnum())
+    return PokemonDetails(
+        id,
+        name,
+        PokemonType.valueOf(types.first().type.name.toValidFormatTypesEnum()),
+        type2,
+        getFlavor(pokedexSelected.gameVersion, pokemonSpecieResponse.flavorTextEntries),
+        BaseStats(
+            stats[0].baseStat,
```

Comment:
los stats puede que no siempre vengan, o sea, por algun error del servidor. seria mejor que sean nullables y los obtengas con `stats.getOrNull(index)`

Reply:
crea una extension de String
`
fun String?.orEmpty(): String = this ?: ""
`

asi no cambias lo del BaseStats a nulls, sino que siguen igual, y solo haces
`stats/getOrNull(0).orEmpty()` usando la extension esa